### DISCUSSION TITLE
can you give me an exmple of how to create a frame around the QR code like this?

### DISCUSSION THREAD

**Discussion by deepesharora** [chillerlan/php-qrcode#181] [General]
@codemasher you've been awesome with the examples you've provided with this repository. 
I'm trying to create something like this with such kind of frames as shown in the pictures. 
https://i.imgur.com/B16LP5r.png

Any guidance or examples you cn provide on how to achieve the same kind of effect with this library?

**Comment by codemasher** [comment]
If you want to do this in SVG, you can start from the examples given in #150, for raster images like png or jpg, you will need to create the frame by yourself in GD or ImageMagick and embed the QR Code in it (i can assist you with the latter in that case).
If you need examples for any of, you can check out the following links:

- http://apex.infogridpacific.com/SVG/svg-tutorial-contents.html
- https://developer.mozilla.org/en-US/docs/Web/SVG
- https://phpimagick.com/Tutorial
- https://www.php.net/manual/en/book.image

**Reply by adi-rebellion** [reply]
Hey! Can you help us with the code how to add text and image below and above an qr-code .
We have trying to use https://github.com/codemasher/nova-qrcode and going through the examples too, but didn't able to achive 
Help us with the code extending inside the generator.php file. (We are able to run this repo locally and able to generate normal qr-codes)

It would be great if you could spare some time and join a 5 min zoom call where you can give a hands-on on what we are stuck with.

**Reply by codemasher** [reply]
Hi, no zoom calls, sorry. I just replied to your email, but i can repeat that here:

> please check the examples [over here](https://github.com/chillerlan/php-qrcode/tree/main/examples)
>The “[GD Image with logo](https://github.com/chillerlan/php-qrcode/blob/main/examples/imageWithLogo.php)” and “[GD image with text](https://github.com/chillerlan/php-qrcode/blob/main/examples/imageWithText.php)” examples are probably closest to what you want to achieve.

**Reply by adi-rebellion** [reply]
Hey @codemasher we were able to generate the qr-code as per our requirements(ofcourse with your guidence), but the only thing  we are not able to is increase or play with the font size of the text, please advise.
![chillerlan](https://user-images.githubusercontent.com/32679846/219782932-3542f6b7-5bf9-4cc8-8935-4a2f99207864.png)

**Reply by codemasher** [reply]
Hey, great that the example worked for you! To be honest, i don't have much experience with drawing fonts in images and from what it seems the GD image library is rather limited in that regard. There is [`imagefontwidth()`](https://www.php.net/manual/en/function.imagefontwidth) and [`imagefontheight()`](https://www.php.net/manual/en/function.imagefontheight.php) and the possible values for each of them range from 1 to 5.

If that isn't enough and you would like to try something different, you can start off from one of the ImageMagick examples. Imagemagick is more flexible in that regard and also allows loading TrueType fonts and add effects, see this demo: https://phpimagick.com/Tutorial/fontEffect

**Reply by codemasher** [reply]
Wait, i just realized that there's only the basic ImageMagick example. I may add another one with text and logo, similar to the existing GD examples, but not today. (:

**Comment by deepesharora** [comment]
Ok. We'll try playing around with imagemagick












---

NOTICE - CONFIDENTIAL INFORMATION
The information in this communication is privileged and strictly 
confidential. It is intended solely for the use of the individual or entity 
named above. If the reader of this message is not the intended recipient, 
or the employee or agent responsible to deliver it to the intended 
recipient, any dissemination, distribution, copying or other use of the 
information contained in this communication is strictly prohibited. If you 
have received this communication in error, please first notify the sender 
immediately and then delete this communication from all data storage 
devices and destroy all hard copies.

On February 17, 2023 15:06:11 smiley ***@***.***> wrote:

Wait, i just realized that there's only the Basic ImageMagick example. I 
may add another one with image and logo, similar to the existing GD 
examples, but not today. (:
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: 
***@***.***>
